server.go

```diff
@@ -1379,6 +1383,9 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 	}
 	opts := &transport.Options{Last: true}
 
+	if stream.SendCompress() != sendCompressorName {
```

Comment:
Nit: add a comment here saying the server handler could have set a new compressor using a call to SetSendCompressor(), and if that is the case, then we need to use that instead.